post_office/utils.py

```diff
@@ -138,14 +138,22 @@ def parse_emails(emails):
     return emails
 
 
-def cleanup_expired_mails(cutoff_date, delete_attachments=True):
+def cleanup_expired_mails(cutoff_date, delete_attachments=True, batch_size=10000):
     """
     Delete all emails before the given cutoff date.
     Optionally also delete pending attachments.
     Return the number of deleted emails and attachments.
     """
     expired_emails = Email.objects.only('id').filter(created__lt=cutoff_date)
-    emails_count, _ = expired_emails.delete()
+    total_deleted_emails = 0
+    while expired_emails:
+        email_ids = expired_emails.values_list('id', flat=True)[:batch_size]
```

Comment:
dirubah jadi list, kalo nggak jadi nested query

Reply:
> change to list first, i think this will make nested query

I think I can make it into a list of email ids start from the expired_emails variable, WDYT?

Reply:
ah i think thats better, so no n-query executing `values_list` in looping